tokio/src/sync/once_cell.rs

```diff
@@ -149,6 +149,38 @@ impl<T> OnceCell<T> {
         }
     }
 
+    /// Creates a new `OnceCell` that contains the provided value.
+    ///
+    /// This is equivalent to `<OnceCell<T> as From<T>>::from(value)`.
+    ///
+    /// # Example
+    ///
+    /// ```
+    /// use tokio::sync::OnceCell;
+    ///
+    /// static ONCE: OnceCell<u32> = OnceCell::from_value(1);
+    ///
+    /// async fn get_global_integer() -> &'static u32 {
+    ///     ONCE.get_or_init(|| async {
+    ///         1 + 1
+    ///     }).await
+    /// }
+    ///
+    /// #[tokio::main]
+    /// async fn main() {
+    ///     let result = get_global_integer().await;
+    ///     assert_eq!(*result, 1);
+    /// }
+    /// ```
+    #[cfg(not(all(loom, test)))]
+    pub const fn from_value(value: T) -> Self {
```

Comment:
I think it makes more sense to have a `const_new_with` constructor instead.

Reply:
> I think it makes more sense to have a `const_new_with` constructor instead.

We can't due to https://github.com/rust-lang/rust/issues/73255 , doing so would fail with `Drop for Option<T>` is not available in const.

Reply:
Hmm, it might still make sense to use that name. The const constructors have fewer features due to tracing integration, so I'd like to nudge people who don't need const to the other constructors.

Reply:
That sounds reasonable, I've renamed it to `const_new_with`.

Reply:
I guess the only remaining annoyance is that the const and non-const versions now differ. But maybe it's fine.